RevenueCatUI/Scripts/PaywallsBehaviour.cs

```diff
@@ -1,32 +1,180 @@
+
+        private bool isPresenting = false;
+
+        public string OfferingIdentifier
+        {
+            get => offeringIdentifier;
+            set => offeringIdentifier = value;
+        }
+
+        public bool DisplayCloseButton
+        {
+            get => displayCloseButton;
+            set => displayCloseButton = value;
+        }
+
+        public string RequiredEntitlementIdentifier
+        {
+            get => requiredEntitlementIdentifier;
+            set => requiredEntitlementIdentifier = value;
+        }
+
         /// <summary>
-        /// Presents a paywall configured in the RevenueCat dashboard.
+        /// Presents the paywall with the configured options.
+        /// Can be called from Unity UI buttons or programmatically.
         /// </summary>
-        /// <param name="options">Options for presenting the paywall.</param>
-        /// <returns>A <see cref="PaywallResult"/> describing the outcome.</returns>
-        public async Task<PaywallResult> PresentPaywall(PaywallOptions options = null)
+        public async void PresentPaywall()
         {
-            return await PaywallsPresenter.Present(options);
+            if (isPresenting)
+            {
+                Debug.LogWarning("[RevenueCatUI] Paywall is already being presented.");
+                return;
+            }
+
+            isPresenting = true;
+
+            try
+            {
+                var options = CreateOptions();
+                PaywallResult result;
+
+                if (!string.IsNullOrEmpty(requiredEntitlementIdentifier))
+                {
+                    result = await PaywallsPresenter.PresentIfNeeded(requiredEntitlementIdentifier, options);
+                }
+                else
+                {
+                    result = await PaywallsPresenter.Present(options);
+                }
+
+                HandleResult(result);
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"[RevenueCatUI] Exception in PaywallsBehaviour: {e.Message}");
+                HandleResult(PaywallResult.Error);
+            }
+            finally
+            {
+                isPresenting = false;
+            }
         }
 
         /// <summary>
         /// Presents a paywall only if the user does not have the specified entitlement.
         /// </summary>
-        /// <param name="requiredEntitlementIdentifier">Entitlement identifier to check before presenting.</param>
-        /// <param name="options">Options for presenting the paywall.</param>
-        /// <returns>A <see cref="PaywallResult"/> describing the outcome.</returns>
-        public async Task<PaywallResult> PresentPaywallIfNeeded(string requiredEntitlementIdentifier, PaywallOptions options = null)
+        /// <param name="entitlementIdentifier">Entitlement identifier to check before presenting</param>
+        public async void PresentPaywallIfNeeded(string entitlementIdentifier)
```

Comment:
Hmm I'm just wondering, since whether using `PresentPaywall` or this one depends on the presence of a parameter, I'm wondering if we should unify them into a single method that just uses one or the other depending on that parameter...

Reply:
I think that makes sense yes! I did this in case wanted to call this directly via code. But they can still do it using the setter, or the static version